Ideas-2025.md

```diff
@@ -95,4 +102,218 @@ Specifically, we would be working toward accomplishing the following:
 **Coding Mentors**<br>
 [Walter Bender](https://github.com/walterbender/)<br>
 **Assisting Mentors**<br>
-[Devin Ulibarri](https://github.com/pikurasa/)
\ No newline at end of file
+[Devin Ulibarri](https://github.com/pikurasa/)
+
+------------
+
+## Add an AI-assistant to the Write Activity
+
+**Prerequisites**<br>
+ - Experience with Python
+ - Experience with Sugar activities
+ - Experience with LLMs/Chatbots
+
+**Description**<br>
+
+Sugar pioneered peer editing in its Write activiy. However, the Write
+Activity has never had any serious support for grammar correction
+(just spell check) and none of the more recent developments around
+AI-assisted writing. The goal of this project is to add AI-assistance
+to the writing process: both in the form of providing feedback as to
+what has been written and making suggestions as to what might be
+written.
+
+The challenge will be both in terms of workflow integration and UX.
+
+**Project Length**<br>
+
+**350** hours
+
+**Difficulty**<br>
+
+**High**
+
+**Coding Mentors**<br>
+[Walter Bender](https://github.com/walterbender/)
+[Ibiam Chihurumnaya](https://github.com/chimosky/)
+
+**Assisting Mentors**<br>
+
+------------
+
+## Refactor the Infoslicer Activity to generate plain-language summaries
+
+**Prerequisites**<br>
+ - Experience with Python
+ - Experience with Sugar activities
+ - Experience with LLMs/Chatbots
+
+**Description**<br>
+
+The Infoslicer Activity is designed to help teachers extract content from the Wikipedia in order to c    reate lesson plans. This is currently a manual, extractive process. It is well suited to generative A    I. The goal would be to have a teacher type in a theme for a lesson and have the AI create a simple l    esson plan, which the teacher can then edit.
+
+The biggest challenge to summarization using generative AI is hallucinations. A work-around for this     is to include a validation step that surfaces evidence (or lack of evidence) for each assertion in th    e lesson plan. This will introduce some workflow and UX challenges.
+
+**Project Length**<br>
+
+**350** hours
+
+**Difficulty**<br>
+
+**High**
+
+**Coding Mentors**<br>
+[Walter Bender](https://github.com/walterbender/)
+[Ibiam Chihurumnaya](https://github.com/chimosky/)
+
+**Assisting Mentors**<br>
+
+------------
+
+ ## Refactor the chatbot in the Speak Activity to use AI
+
+**Prerequisites**<br>
+ - Experience with Python
+ - Experience with Sugar activities
+ - Experience with LLMs/Chatbots
+
+**Description**<br>
+
+The [Speak Activity](https://github.com/sugarlabs/speak) is one of most popular Sugar activities. It     allows
+someone just beginning to familiarize themselves with reading to
+interact with synthetic speech. It has both chat and chatbot
+capabilities, so that learners can share what they type with others,
+often using invented spelling. It would be a nice improvement if there
```

Comment:
In addition to this, if it's not too outside of the scope, we could use a (local?) neural network to improve audio for the voice to sound more natural. Perhaps this could be a separate project, either for this year or for a future year.

Reply:
I think this is better as a future project, because as a separate project it'll be dependent on this idea.